content/strategy-goals/strategy/batch-changes/index.md

```diff
@@ -59,82 +59,67 @@ We have the overarching goal to [level up our enterprise-ready features](../inde
 - Creating changesets locally takes too long to be practical for very large-scale changes.
 - Navigation in the UI when there are 100s of batch changes and 10,000s of changesets is cumbersome.
 - Batch Changes's permission model is minimalistic and insufficient for some large enterprises.
+- Supporting running batch changes in large monorepos
 
 To drive enterprise success and stickiness, we need to:
 
 - Deliver a great experience in creating changesets at **enterprise scale** (10,000s repositories), resulting in more users onboarding batch changes.
-- Make sure that iterating on batch changes has the **lowest cycle time possible**, and that the debugging experience is smooth.
+- Support enterprise requirements in terms of permissions and controls.
 - Enable our users to make sense of a large number of batch changes and changesets.
 
 ### Get changesets merged
 
 The key [success metric](../../../departments/engineering/teams/batch-changes/metrics.md) for batch changes is the number of changesets that get _merged_. In practice, once a user has applied a batch change, they need to spend time tracking and following up with downstream repository owners so that they merge the batch change's changesets. The larger the batch change, the more time-consuming this can be. We want to improve how easy it is for users to get large batch changes over the finish line, as measured by time-to-merge and batch changes merge rate.
 
-### Time-to-value
+### Time to value
 
-Batch Changes is a powerful product with a steep learning curve. The main friction point is that Batch Changes requires users to write **code rewrite steps** (the components of a batch spec that define a specific change to be executed), either using existing code rewrite tools (sed, comby, etc) or writing code from scratch. Most developers are unfamiliar with such tooling, and have a hard time onboarding to Batch Changes.
+Batch Changes is a powerful product with a steep learning curve. The two main friction points are:
 
-We want to decrease time to value and make it easier for any developer to get value out of Batch Changes. We will measure success here by tracking:
+- iterating on a batch change has a slow cycle time
+- Batch Changes requires users to write **code rewrite steps** (the components of a batch spec that define a specific change to be executed), either using existing code rewrite tools (sed, comby, etc) or writing code from scratch. Most developers are unfamiliar with such tooling, and have a hard time onboarding to Batch Changes.
 
-- time to value ([#32664](https://github.com/sourcegraph/sourcegraph/issues/32664))
-- the batch changes success rate (number of batch changes that get applied / number of batch changes that stay in preview)
+We are pursuing three approaches here:
```

Comment:
I don't disagree with the "sandbox" approach, however to combat this problem there are other things that we can (and should) do to the actual experience like reviewing/standardising copy and naming conventions. We can also experiment with the layouts of each step to convey more confidence to the user.

We can also provide a better onboarding experience, so that the user understand better each step.

Anyway, I just don't want to be too focused on one possible solution

Reply:
I am adding this problem, leaving potential solutions open.